Fixes mentions of `contrib` and replaces with `main` (#7066)
umbraco/UmbracoDocs: 10/umbraco-cms/reference/cache/README.md

In normal environments caching seems to be a pretty standard concept. If you are a package developer or developer who is going to publish a codebase to a load balanced environment then you need to be aware of how to invalidate your cache properly, so that it works in load balanced environments. If it is not done correctly then your package and/or codebase will not work the way that you would expect in a load balanced scenario.

> **If you are caching business logic data that changes based on a user's action in the backoffice and you are not using an **_**ICacheRefresher**_** then you will need to review your code and update it based on the below documentation.**

Reviewer: ```suggestion
**If you're caching business logic based on backoffice user actions without using an _ICacheRefresher_, review and update your code using the documentation below.**
```

* `ICacheRefresher<T>` - this inherits from `ICacheRefresher` and provides a set of strongly typed methods for cache invalidation. This is useful when executing the method to invoke the cache refresher, when you have the instance of the object already since this avoids the overhead of retrieving the object again.
  * `void Refresh(T instance);` - this would invalidate/refresh a single cache for the specified object.
  * `void Remove(T instance);` - this would invalidate a single cache for the specified object.

> * `IJsonCacheRefresher` - this inherits from `ICacheRefresher` but provides more flexibility if you need to invalidate cache based on more complex scenarios (e.g. the [MemberGroupCacheRefresher](https://github.com/umbraco/Umbraco-CMS/blob/v13/main/src/Umbraco.Core/Cache/MemberGroupCacheRefresher.cs)).

Reviewer: This link does not work. It says 404 - Page not found.

Author: Thanks, I replaced it with a permalink from GitHub instead. The linked file belonged to V10, so it does not exist anymore.

Reviewer: Perfect, thanks 😊

> * `IJsonCacheRefresher` - this inherits from `ICacheRefresher` but provides more flexibility if you need to invalidate cache based on more complex scenarios (e.g. the [MemberGroupCacheRefresher](https://github.com/umbraco/Umbraco-CMS/blob/v13/main/src/Umbraco.Core/Cache/MemberGroupCacheRefresher.cs)).

Reviewer: ```suggestion
* `IJsonCacheRefresher` - this inherits from `ICacheRefresher` but provides more flexibility if you need to invalidate cache based on more complex scenarios (for example, the [MemberGroupCacheRefresher](https://github.com/umbraco/Umbraco-CMS/blob/v13/main/src/Umbraco.Core/Cache/MemberGroupCacheRefresher.cs)).
```

  * `void Refresh(string jsonPayload)` - Invalidates/refreshes any cache based on the information provided in the JSON. The JSON value is any value that is used when executing the method to invoke the cache refresher.

> There are several examples of `ICacheRefresher`'s in the core: https://github.com/umbraco/Umbraco-CMS/tree/v13/main/src/Umbraco.Core/Cache

Reviewer: ```suggestion
There are a couple of examples of `ICacheRefresher's` in the [core](https://github.com/umbraco/Umbraco-CMS/tree/v13/main/src/Umbraco.Core/Cache).
```
